pkg/querier/tripperware/queryrange/results_cache.go

```diff
@@ -295,6 +295,11 @@ func (s resultsCache) shouldCacheResponse(ctx context.Context, req tripperware.R
 	if !s.isOffsetCachable(ctx, req) {
 		return false
 	}
+	if res, ok := r.(*tripperware.PrometheusResponse); ok {
+		if len(res.Warnings) > 0 {
```

Comment:
Should this explicitly check for partial response warning only? Engine could return other warnings which can be cache-able.

Reply:
True, will fix it

Reply:
Fixed